Distance_2/include/CGAL/squared_distance_2_1.h

```diff
@@ -197,7 +197,109 @@ namespace internal {
   typename K::FT
   squared_distance(const typename K::Segment_2 &seg1,
 		   const typename K::Segment_2 &seg2,
-		   const K& k)
+		   const K& k,
+                   const Cartesian_tag&)
+  {
+    typedef typename K::RT RT;
+    typedef typename K::FT FT;
+    bool crossing1, crossing2;
+    RT c1s, c1e, c2s, c2e;
+    if (seg1.source() == seg1.target())
+      return internal::squared_distance(seg1.source(), seg2, k);
+    if (seg2.source() == seg2.target())
+      return internal::squared_distance(seg2.source(), seg1, k);
+    
+    Orientation o1s = orientation(seg2.source(), seg2.target(), seg1.source());
+    Orientation o1e = orientation(seg2.source(), seg2.target(), seg1.target());
+    Orientation o2s = orientation(seg1.source(), seg1.target(), seg2.source());
+    Orientation o2e = orientation(seg1.source(), seg1.target(), seg2.target());
+    if (o1s == RIGHT_TURN) {
+      crossing1 = (o1e != RIGHT_TURN);
+    } else {
+      if (o1e != LEFT_TURN) {
+	if (o1s == COLLINEAR && o1e == COLLINEAR)
+	  return internal::squared_distance_parallel(seg1, seg2, k);
+	crossing1 = true;
+      } else {
+	crossing1 = (o1s == COLLINEAR);
+      }
+    }
+    if (o2s == RIGHT_TURN) {
+      crossing2 = (o2e != RIGHT_TURN);
+    } else {
+      if (o2e != LEFT_TURN) {
+	if (o2s == COLLINEAR && o2e == COLLINEAR)
+	  return internal::squared_distance_parallel(seg1, seg2, k);
+	crossing2 = true;
+      } else {
+	crossing2 = (o2s == COLLINEAR);
+      }
+    }
+
+    c1s = CGAL::abs(wcross(seg2.source(), seg2.target(), seg1.source(), k));
+    c1e = CGAL::abs(wcross(seg2.source(), seg2.target(), seg1.target(), k));
+    c2s = CGAL::abs(wcross(seg1.source(), seg1.target(), seg2.source(), k));
+    c2e = CGAL::abs(wcross(seg1.source(), seg1.target(), seg2.target(), k));
```

Comment:
There is no need to compute all four values. For example, the first case only uses `c2s` and `c2e`.

Reply:
changed.